model/src/main/scala/ch.epfl.scala.index.model/release/BackendVersion.scala

```diff
@@ -0,0 +1,96 @@
+package ch.epfl.scala.index.model
+package release
+
+import fastparse.NoWhitespace._
+import fastparse._
+
+/**
+ * A [[BackendVersion]] is the binary version of the compiler backend.
+ * It can be either a [[ScalaBackend]] or a [[DottyBackend]]
+ **/
+sealed trait BackendVersion {
+
+  /**
+   * When indexing, all dotty versions are regrouped under the 'dotty' keyword.
+   * That keyword is called the family of the version.
+   * @return family
+   */
+  def family: String
+  def render: String
+
+  /**
+   * This method is used to discard deprecated or ill-formated versions.
+   * It returns true if the version is relevant and false otherwise.
+   * @return isValid
+   */
+  def isValid: Boolean
+}
+
+final case class ScalaBackend(version: BinaryVersion) extends BackendVersion {
+  def family: String = toString()
+  def render: String = s"scala $version"
+  def isValid: Boolean = ScalaBackend.isValid(version)
+  override def toString: String = version.toString()
+}
+
+final case class DottyBackend(version: BinaryVersion) extends BackendVersion {
+  def family = "dotty"
+  def render: String = s"dotty $version"
+  def isValid: Boolean = DottyBackend.isValid(version)
+  override def toString: String = version.toString()
+}
+
+object ScalaBackend {
+  val `2.10`: ScalaBackend = ScalaBackend(MinorBinary(2, 10))
+  val `2.11`: ScalaBackend = ScalaBackend(MinorBinary(2, 11))
+  val `2.12`: ScalaBackend = ScalaBackend(MinorBinary(2, 12))
+  val `2.13`: ScalaBackend = ScalaBackend(MinorBinary(2, 13))
+
+  private val stableBinaryVersions =
+    Set(`2.10`.version, `2.11`.version, `2.12`.version, `2.13`.version)
+
+  def isValid(version: BinaryVersion): Boolean =
+    stableBinaryVersions.contains(version)
+
+  def Parser[_: P]: P[ScalaBackend] = {
+    BinaryVersion.Parser
+      .filter(isValid)
+      .map(ScalaBackend(_))
+  }
+}
+
+object DottyBackend {
+  val `0.23`: DottyBackend = DottyBackend(MinorBinary(0, 23))
+
+  def isValid(version: BinaryVersion): Boolean =
+    version match {
+      case MinorBinary(major, minor) if major == 0 && minor < 30 => true
```

Comment:
```suggestion
      case MinorBinary(major, minor) if major == 0 => true
```
All 0.x versions are Dotty version. There's no artificial cap at 30.

Reply:
Right but I definetely dont want to interpret this sbt plugin version _0.31_2.12 version as a plugin of sbt  2.12 compiled with Dotty 0.31. Would a `isValid` sbt version that discard sbt `2.12` be better?

Reply:
Yes I think it is actually